central/sensor/service/connection/connection_impl.go

```diff
@@ -576,6 +579,28 @@ func (c *sensorConnection) Run(ctx context.Context, server central.SensorService
 		}
 	}
 
+	if connectionCapabilities.Contains(centralsensor.SensorReconciliationOnReconnect) {
+		// Sensor is capable of doing the reconciliation by itself if receives the hashes from central.
+		// Central should only attempt to do that if sensor is reconnecting, otherwise Central should
+		// do the reconciliation itself, since events might have been missed.
+		if c.sensorHello.GetSensorState() == central.SensorHello_RECONNECT {
+			log.Info("Sensor reconnecting: Reconciliation will be done in Sensor")
+			c.sensorEventHandler.disableReconciliation()
+			// Send hashes to sensor
+			successfulHashes := c.hashDeduper.GetSuccessfulHashes()
+			err := server.Send(&central.MsgToSensor{Msg: &central.MsgToSensor_DeduperState{
+				DeduperState: &central.DeduperState{
+					ResourceHashes: successfulHashes,
+				},
+			}})
+			if err != nil {
+				log.Errorf("Sending deduper state to Sensor: %s", err)
```

Comment:
This is effectively catastrophic right? Imagine the case where the hashes exceed the max message size and this is an error. What will Sensor do? How do we recover?

Reply:
That makes sense, thanks for bringing it up. The current idea we have is captured in another ticket ([ROX-19670](https://issues.redhat.com/browse/ROX-19670)). If Central fails to send the deduper state, the reconciliation map should not be closed, and will be to send the `SYNC` events.

I made some changes to the code to not close the reconciliation map and continue with Central reconciliation. Let me know what you think.

Reply:
Also, the exceeding the max message size is a real thing, let's make sure to scale test this